### PR TITLE
Admin add movie

### PR DESCRIPTION
**Pull request by nhat15**
## Related Tickets
- [#51794](https://edu-redmine.sun-asterisk.vn/issues/51794)

## WHAT (optional)
- Change movies in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-08-22 09-44-00
Screenshot from 2022-08-22 09-44-25

### CODE REVIEW

**Comment by nhat15** [comment]
ready

File: app/models/movie.rb [hunk 1]
```diff
@@ -5,7 +5,21 @@ class Movie < ApplicationRecord
   has_many :ratings, dependent: :destroy
   has_many :shows, dependent: :destroy
   has_one_attached :image
+  MOVIE_ATTR = %i(title description release_date duration language cast
+director genre_id).freeze
+
+  validates :title, presence: true,
+length: {maximum: Settings.movie.name.max_length}, uniqueness: true
```
Comment (huongnt-2545) [inline app/models/movie.rb]: sai indent

File: app/models/movie.rb [hunk 2]
```diff
@@ -5,7 +5,21 @@ class Movie < ApplicationRecord
   has_many :ratings, dependent: :destroy
   has_many :shows, dependent: :destroy
   has_one_attached :image
+  MOVIE_ATTR = %i(title description release_date duration language cast
+director genre_id).freeze
```
Comment (huongnt-2545) [inline app/models/movie.rb]: sai indent

File: app/models/movie.rb [hunk 3]
```diff
@@ -5,7 +5,21 @@ class Movie < ApplicationRecord
...
+  validates :title, presence: true,
+length: {maximum: Settings.movie.name.max_length}, uniqueness: true
+  validates :description, presence: true,
+length: {maximum: Settings.movie.content.max_length}
+  validates :release_date, presence: true,
+  date: {after: proc{Time.zone.now}}, on: :save
+  validates :duration, presence: true
+  validates :language, presence: true
+  validates :cast, presence: true
+  validates :director, presence: true
```
Comment (huongnt-2545) [inline app/models/movie.rb]: cùng validate `presence: true` thì em gộp lại 1 dòng nhé

File: app/views/admin/layouts/_sidebar.html.erb [hunk 4]
```diff
@@ -1,39 +1,38 @@
...
-            <i class='fa fa-credit-card nav_icon'></i>
-            <span class="nav_name"><%= t ".revenue" %></span>
-          </a>
-        </div>
-      </div>
-      <a href="#" class="nav_link"> <i class='fa fa-sign-out'></i> <span class="nav_name"><% t ".signout" %></span> </a>
-    </nav>
-  </div>
-</body>
+    <header class="header" id="header">
```
Comment (huongnt-2545) [inline app/views/admin/layouts/_sidebar.html.erb]: check lại indent trong file này em nhé
hiện indent = 4 spaces

File: app/views/admin/movies/_movie.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  <th><%= movie.language %></th>
+  <th><%= movie.cast %></th>
+  <th><%= movie.director %></th>
+  <th><%= movie.status %></th>
+  <th>
+    <%= link_to edit_admin_movie_path(movie) do %>
+      <span class="fa fa-pencil"></span>
+    <% end %>
+    <%= link_to admin_movie_path(movie), method: :delete,
+          data: { confirm: "Are you sure?"} do %>
```
Comment (huongnt-2545) [inline app/views/admin/movies/_movie.html.erb]: I18n

File: app/views/admin/movies/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      <th><%= t ".status" %></th>
+      <th><%= t ".action" %></th>
+    </tr>
+  </thead>
+  <tbody>
+    <%= render partial: "movie", collection: @movies %>
+  </tbody>
+</table>
+
+<%== pagy_bootstrap_nav @pagy%>
```
Comment (huongnt-2545) [inline app/views/admin/movies/index.html.erb]: end file

**Comment by nhat15** [comment]
ready

File: app/controllers/admin/movies_controller.rb [hunk 7]
```diff
@@ -0,0 +1,63 @@
+class Admin::MoviesController < Admin::AdminController
+  before_action :find_movie, only: %i(edit update destroy)
+  before_action :load_genre, only: %i(new edit update create)
```
Comment (ghost) [inline app/controllers/admin/movies_controller.rb]: dùng except đi em

File: app/views/admin/movies/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+<h1><%= t ".movie" %></h1>
+<%= link_to new_admin_movie_path, class: "new-genre" do %>
+  <span class="fa fa-plus-circle circle"></span>
+  <%= t ".create" %>
+<% end %>
+<table>
+  <thead>
+    <tr>
+      <th><%= t ".id"%></th>
```
Comment (ghost) [inline app/views/admin/movies/index.html.erb]:
```suggestion
      <th><%= t ".id" %></th>
```

**Review by ghost — COMMENTED** [review]
cmt nhỏ

**Comment by nhat15** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**